safe/gui/tools/minimum_needs/needs_calculator_dialog.py

```
@@ -61,148 +100,57 @@ def __init__(self, parent=None):
+        # count each minimum needs for every features
+        for needs in minimum_needs_post_processors:
+            run_single_post_processor(input_layer, needs)
 
     def accept(self):
         """Process the layer and field and generate a new layer.
 
         .. note:: This is called on OK click.
 
         """
-        index = self.cboFields.currentIndex()
-        field_name = self.cboFields.itemData(
-            index, QtCore.Qt.UserRole)
-
-        index = self.cboPolygonLayers.currentIndex()
-        layer_id = self.cboPolygonLayers.itemData(
-            index, QtCore.Qt.UserRole)
-        # noinspection PyArgumentList
-        layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
-
-        file_name = layer.source()
+        # create memory layer
+        output_layer = (
+            '%s_minimum_needs' % self.layer.currentLayer().name())
+        output_layer = create_memory_layer(
+            output_layer,
+            self.layer.currentLayer().geometryType(),
+            self.layer.currentLayer().crs(),
+            self.layer.currentLayer().fields())
+
+        # monkey patching input layer to make it work with
+        # prepare vector layer function
+        temp_layer = self.layer.currentLayer()
+        temp_layer.keywords = {'layer_purpose': 'aggregation'}
+
+        # add keywords to output layer
+        copy_layer(temp_layer, output_layer)
+        output_layer.keywords['layer_purpose'] = 'aggregation'
+        output_layer.keywords['inasafe_fields'] = (
+            {'displaced_field': self.displaced.currentField(),
+             'aggregation_name_field': self.aggregation_name.currentField()})
+
+        # remove unnecessary fields & rename inasafe fields
+        _rename_remove_inasafe_fields(output_layer)
 
         try:
-            output_layer = self.minimum_needs(input_layer, field_name)
-        except ValueError:
+            self.minimum_needs(output_layer)
+        except:
             return
```

We don't want to silently fail if there is an error. You should display an error message and log it using the LOGGER.